plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineStandardWizardPage.java

```diff
@@ -0,0 +1,172 @@
+    workspaceProjectDirectoryButtonPosition.horizontalSpan = 3;
+    workspaceProjectDirectoryButton.setLayoutData(workspaceProjectDirectoryButtonPosition);
+
+    customProjectDirectoryButton = new Button(projectDirectoryGroup, SWT.RADIO);
+    customProjectDirectoryButton.setSelection(false); // not by default
+    customProjectDirectoryButton.setText("Create new project in:");
+    customProjectDirectoryButton.addSelectionListener(directorySwitcher);
+    GridData customProjectDirectoryButtonPosition = new GridData();
+    customProjectDirectoryButtonPosition.horizontalAlignment = GridData.FILL;
+    customProjectDirectoryButtonPosition.grabExcessHorizontalSpace = true;
+    customProjectDirectoryButtonPosition.horizontalSpan = 3;
+    customProjectDirectoryButton.setLayoutData(customProjectDirectoryButtonPosition);
+
+    Label projectDirectoryLabel = new Label(projectDirectoryGroup, SWT.NONE);
+    projectDirectoryLabel.setText("Directory:");
+
+    projectDirectoryField = new Text(projectDirectoryGroup, SWT.BORDER);
+    projectDirectoryField.setEnabled(false);
+    GridData projectDirectoryFieldPosition = new GridData();
+    projectDirectoryFieldPosition.horizontalAlignment = GridData.FILL;
+    projectDirectoryFieldPosition.grabExcessHorizontalSpace = true;
+    projectDirectoryField.setLayoutData(projectDirectoryFieldPosition);
+
+    projectDirectoryBrowseButton = new Button(projectDirectoryGroup, SWT.NONE);
+    projectDirectoryBrowseButton.setEnabled(false);
+    projectDirectoryBrowseButton.setText("Browse...");
+    projectDirectoryBrowseButton.addSelectionListener(new ProjectDirectoryPicker());
+  }
+  
+  private final class ProjectDirectoryPicker extends SelectionAdapter {
+    @Override
+    public void widgetSelected(SelectionEvent event) {
+      DirectoryDialog dialog = new DirectoryDialog(getShell(), SWT.OPEN);
+      dialog.setMessage("Choose a directory for the project contents:");
+      String userChoice = dialog.open();
+      if (userChoice != null) {
+        projectDirectoryField.setText(userChoice);
+      }
+    }
+  }
+
+  private final class PageValidator implements ModifyListener {
+    @Override
+    public void modifyText(ModifyEvent event) {
+      // todo add checks for project ID and directory 
+      // todo add error messages
+      boolean complete = JavaPackageValidator.validate(javaPackageField.getText()) 
+          && EclipseProjectNameValidator.validate(eclipseProjectNameField.getText());
+      setPageComplete(complete);
+    }
+  }  
+  
+  private final class DirectorySwitcher extends SelectionAdapter {
+    @Override
+    public void widgetSelected(SelectionEvent event) {
+      projectDirectoryBrowseButton.setEnabled(event.widget == customProjectDirectoryButton);
+      projectDirectoryField.setEnabled(event.widget == customProjectDirectoryButton);
+    }
+  }
+
+  AppEngineStandardProjectConfig getAppEngineStandardProjectConfig() {
```

Comment:
Something along that line.  What I've found easiest to manage is for the `Wizard` instance to configure the `WizardPage` with a particular model object (like `AppEngineStandardProjectConfig`), and the `WizardPage` changes that model object.  The `Wizard` knows that the `WizardPage` is in a consistent state via `WizardPage#isPageComplete()`.

So essentially the `WizardPage` is a view, the `Wizard` is the controller.